torch/distributed/device_mesh.py

```diff
@@ -488,6 +325,162 @@ def __init__(
                     rank_coords[0].tolist() if rank_coords.size(0) > 0 else None
                 )
 
+        def _maybe_create_backend(
```

Comment:
as a refactoring thing, it might be nice to extract out the 'create backend' part of this as a separate util, so this function is focused on caching logic.

also i'd rename it to `get_or_create` not `maybe_create`